### PR TITLE
feat(expo-cli): credentials manager for ios

### PR DESCRIPTION
**Pull request by wkozyra95**

### CODE REVIEW

File: packages/expo-cli/src/credentials/api.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  IosAppCredentials,
+  IosPushCredentials,
+  IosDistCredentials,
+} from './credentials';
+
+
+export class IosApi {
+  api: ApiV2;
+  credentials: IosCredentials;
+  shouldRefetch: boolean = true
```
Comment (dsokal) [inline packages/expo-cli/src/credentials/api.ts]:
```suggestion
  shouldRefetch: boolean = true;
```

File: packages/expo-cli/src/credentials/views/IosDistCredentials.ts [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+      const { confirm } = await prompt(question);
+      if (!confirm) {
+        log('Aborting update process');
+        return null;
+      }
+    }
+
+    const newDistCert = await this.provideOrGenerate(ctx);
+    await ctx.ensureAppleCtx();
+    const updatedUserCredentials =await ctx.ios.updateDistCert(selected.id, {
```
Comment (dsokal) [inline packages/expo-cli/src/credentials/views/IosDistCredentials.ts]:
```suggestion
    const updatedUserCredentials = await ctx.ios.updateDistCert(selected.id, {
```

File: packages/expo-cli/src/commands/credentials.ts [hunk 3]
```diff
@@ -2,40 +2,37 @@ import {
   runCredentialsManager,
```
[the remark this answers is not in the document]
Comment (wkozyra95) [reply packages/expo-cli/src/commands/credentials.ts]: flag `--platform android|ios` probably will be cleaner

File: packages/expo-cli/src/credentials/actions/list.ts [hunk 4]
```diff
@@ -3,11 +3,94 @@ import uniq from 'lodash/uniq';
 import isEmpty from 'lodash/isEmpty';
```
[the remark this answers is not in the document]
Comment (wkozyra95) [reply packages/expo-cli/src/credentials/actions/list.ts]: in general I agree, but in this particular case I think further separation will complicate things

File: packages/expo-cli/src/appleApi/distributionCert.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  }
+  async create(): Promise<DistCert> {
+    try {
+      const args = ['create', this.ctx.appleId, this.ctx.appleIdPassword, this.ctx.team.id, String(this.ctx.team.inHouse)];
+      return await runAction(travelingFastlane.manageDistCerts, args);
+    } catch (err) {
+      const resultString = get(err, 'rawDump.resultString');
+      if (resultString && resultString.match(/Maximum number of certificates generated/)) {
+        const error = new CommandError('APPLE_DIST_CERTS_TOO_MANY_GENERATED_ERROR');
+        throw error;
```
Comment (dsokal) [inline packages/expo-cli/src/appleApi/distributionCert.ts]:
```suggestion
        throw new CommandError('APPLE_DIST_CERTS_TOO_MANY_GENERATED_ERROR');
```

File: packages/expo-cli/src/appleApi/pushKey.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  async create(name = `Expo Push Notifications Key ${dateformat('yyyymmddHHMMss')}`): Promise<PushKey> {
+    try {
+      const args = ['create', this.ctx.appleId, this.ctx.appleIdPassword, this.ctx.team.id, name];
+      return await runAction(travelingFastlane.managePushKeys, args);
+    } catch (err) {
+      const resultString = get(err, 'rawDump.resultString');
+      if (resultString && resultString.match(/maximum allowed number of Keys/)) {
+        const error = new CommandError('APPLE_PUSH_KEYS_TOO_MANY_GENERATED_ERROR');
```
Comment (dsokal) [inline packages/expo-cli/src/appleApi/pushKey.ts]:
```suggestion
       throw new CommandError('APPLE_PUSH_KEYS_TOO_MANY_GENERATED_ERROR');
```

File: packages/expo-cli/src/commands/credentials.ts [hunk 7]
```diff
@@ -2,40 +2,30 @@ import {
...
-
-  program
-    .command('credentials:manager:ios')
-    .description('Manage your iOS credentials')
-    .asyncAction(async () => {
-      const projectDir = process.cwd();
-      const context = new Context(); // TODO: implement ios part
-      await context.init(projectDir);
-      await runCredentialsManager(context, new SelectAndroidExperience());
+      let mainpage
```
Comment (dsokal) [inline packages/expo-cli/src/commands/credentials.ts]:
```suggestion
      let mainpage;
```

File: packages/expo-cli/src/credentials/context.ts [hunk 8]
```diff
@@ -25,6 +29,21 @@ export class Context {
   get api(): ApiV2 {
     return this._apiClient as ApiV2;
   }
+  get ios(): IosApi {
+    return this._iosApiClient as IosApi
```
Comment (dsokal) [inline packages/expo-cli/src/credentials/context.ts]:
```suggestion
    return this._iosApiClient as IosApi;
```

File: packages/expo-cli/src/credentials/views/IosDistCert.ts [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+      teamId: ctx.appleCtx.team.id,
+      teamName: ctx.appleCtx.team.name,
+    }
+    return await ctx.ios.createDistCert(credentials);
+  }
+
+  async open(ctx: Context): Promise<IView | null> {
+    const distCert = await this.create(ctx);
+
+    log(chalk.green('Succesfully created Distribution Certificate\n'));
```
Comment (dsokal) [inline packages/expo-cli/src/credentials/views/IosDistCert.ts]:
```suggestion
    log(chalk.green('Successfully created Distribution Certificate\n'));
```

**Pull request merged**